MetaMask/metamask-extension
# Capture Sentry errors prior to initialization

## Explanation

Sentry errors captured before/during the wallet initialization are currently not captured because we don't have the controller state yet to determine whether the user has consented.

The Sentry setup has been updated to check the persisted state for whether the user has consented, as a fallback in case the controller state hasn't been initialized yet. This ensures that we capture errors during initialization if the user has opted in.

## Manual Testing Steps

* Opt-in to metrics
* Open dev tools for the background process, then open the network tab
* Set invalid state and reload the extension
  * e.g. run this in an extension browser console:
    ```
    chrome.storage.local.get(({ data, meta }) => chrome.storage.local.set({ data: { ...data, TokenListController: { tokensChainsCache: { '0x1': {} } }}, meta: { version: 76 } }, () => { window.location.reload() }))
    ```
 * See that the Sentry request is sent with the migration error
 * Try again without opting-in to metrics, and see that the request is not sent.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
